dashboard/src/modules/components/OverviewComponent/common-component.jsx

```diff
@@ -157,3 +171,164 @@ export const RenderPagination = (props) => {
     />
   );
 };
+
+export const EditRow = (props) => {
+  return (
+    <div className="pf-c-inline-edit__action pf-m-enable-editable">
+      {!props.item.isEdit ? (
+        <Button
+          variant="plain"
+          onClick={() => props.toggleEdit(props.item.resource_id, true)}
+          icon={<PencilAltIcon />}
+        />
+      ) : (
+        <div>
+          <Button
+            isDisabled={
+              !props.item.isDirty ||
+              !props.item.name ||
+              props.item.name_validated === "error"
+            }
+            onClick={() =>
+              props.saveRowData("datasetName", props.item, props.item.name)
+            }
+            variant="plain"
+            icon={<CheckIcon />}
+          />
+          <Button
+            variant="plain"
+            onClick={() => props.toggleEdit(props.item.resource_id, false)}
+            icon={<TimesIcon />}
+          />
+        </div>
+      )}
+    </div>
+  );
+};
+
+export const DatasetNameInput = (props) => {
+  return (
```

Comment:
There's no need for the function to evaluate to a code block when the only thing the code does is `return` -- it can just evaluate directly to the expression which we would otherwise have returned.

Reply:
Done